include/boost/math/special_functions/detail/bernoulli_details.hpp

```diff
@@ -372,7 +372,7 @@ class bernoulli_numbers_cache
          {
             for(; n; ++start, --n)
             {
-               *out = b2n_asymptotic<T, Policy>(static_cast<typename container_type::size_type>(start * 2U));
+               *out = b2n_asymptotic<T, Policy>(static_cast<int>(start * 2U));
```

Comment:
I have a hunch this is the wrong fix: a Bernoulli number may be too large for type `int` (overflow) and/or the wrong type for the iterator we are copying too - as the OutputIterator is a template parameter.  

Ideally, we would figure out what the `value_type` of the iterator is, and cast to that, but I don't think we can do that in C++20?  Failing that I think you need to look back up the instantiation stack to see what OutputIterator caused the warning, and possibly you may have to live with it (ie it's a genuine warning that you are doing something which may overflow).

Reply:
But the function `b2n_asymptotic(int)` itself accepts an input parameter of type `int`. Would it make sense to refactor that entire thing to accept `std::size_t` or even `std::ptrdiff_t`? The asme is true for `t2n_asymptotic(int)`.

Reply:
Sorry Chris, I completely mis-read that, please ignore my comment, I thought you were casting the result to an int not the argument.  You were quite right!